AndroidSDKCore/src/main/java/com/leanplum/internal/RequestFactory.java

```diff
@@ -21,20 +21,55 @@
 
 package com.leanplum.internal;
 
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
 import java.util.Map;
+import java.util.prefs.Preferences;
 
 public class RequestFactory {
+  private static final String REQUEST_ID_KEY = "LEANPLUM_REQUEST_ID_KEY";
+  private static final int MAX_REQUEST_ID = 99999;
+
   public static RequestFactory defaultFactory;
 
-  public synchronized static RequestFactory getInstance() {
+  private Context context;
+  private int requestId;
+
+  public synchronized static RequestFactory getInstance(Context context) {
     if (defaultFactory == null) {
-      defaultFactory = new RequestFactory();
+      defaultFactory = new RequestFactory(context);
+      defaultFactory.requestId = defaultFactory.loadRequestIdFromDisk();
     }
     return defaultFactory;
   }
 
+  public RequestFactory(Context context) {
+    this.context = context;
+  }
+
   public Request createRequest(
       String httpMethod, String apiMethod, Map<String, Object> params) {
-    return new Request(httpMethod, apiMethod, params);
+    return new Request(httpMethod, apiMethod, params, incrementedRequestId());
+  }
+
+  private int incrementedRequestId() {
+    requestId = (requestId + 1) % MAX_REQUEST_ID;
+    saveRequestIdToDisk(requestId);
+    return requestId;
```

Comment:
do we need the `requestId` to be an integer? As far as we're concerned this just needs to be unique _within_ the same batch that's being submitted to the server and a uuid gives that to us.

Reply:
uuid is the easiest for sure. Kelly is concerned about the costs and wants this to be the smallest possible. This is a balance between uniqueness and smallest payload

Reply:
I'm glad we're considering cost in our implementation and design, but we're talking about a few dollars here and it's not worth sacrificing reliability (using a standard for generating global ids) and maintainability (maintaining the added code, although it's pretty straight forward ) for.